### PR TITLE
feat(as-4325): add formatting for appeal deadline 

### PR DESCRIPTION
**Pull request by demzzydev**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-4325

## Description of change
adding formatting for appeal deadline

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/e2e-tests/cypress/integration/eligibility/householder-planning/date-decision-due/date-decision-due-steps.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { goToAppealsPage } from '../../../../support/common/go-to-page/goToAppea
 import { selectPlanningApplicationType } from '../../../../support/eligibility/planning-application-type/select-planning-application-type';
 import { selectSiteOption } from '../../../../support/eligibility/appellant-selects-the-site/select-site-option';
 import { selectPlanningApplicationDecision } from '../../../../support/eligibility/granted-or-refused-application/select-planning-application-decision';
+import {clickContinueButton} from "../../../../support/common/clickContinueButton";
 import {
   getDateDecisionDueDay,
   getPlanningApplicationDecisionError,
```
Comment (swdpcomputing) [inline packages/e2e-tests/cypress/integration/eligibility/householder-planning/date-decision-due/date-decision-due-steps.js:18]: Does anything need to happen with the commented out lines 120 & 122?

File: packages/forms-web-app/src/controllers/full-appeal/decision-date.js [hunk 2]
```diff
@@ -35,6 +35,8 @@ exports.postDecisionDate = async (req, res) => {
   );
 
   if (isBefore(enteredDate, add(new Date(todaysDate), { months: -6 }))) {
```
Comment (nssdesikan) [inline packages/forms-web-app/src/controllers/full-appeal/decision-date.js:37]: Value should not be hardcoded

File: packages/forms-web-app/src/controllers/full-appeal/decision-date.js [hunk 3]
```diff
@@ -35,6 +35,8 @@ exports.postDecisionDate = async (req, res) => {
...
+    const deadlineDate = add(enteredDate, { months: 6 });
```
Comment (nssdesikan) [inline packages/forms-web-app/src/controllers/full-appeal/decision-date.js:38]: Value should not be hardcoded

**Review by nssdesikan — CHANGES_REQUESTED** [review]
Its nothing to do with this PR but the view of shutter page is hard coded with 6 months. This should not be hard coded

File: packages/e2e-tests/cypress/integration/eligibility/householder-planning/date-decision-due/date-decision-due-steps.js (hunk 1 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/eligibility/householder-planning/date-decision-due/date-decision-due-steps.js:18]: @demzzydev Can you please un comment them now that we can check the date format?

**Comment by shepda** [comment]
Just discussed with @nssdesikan that to format the date on the shutter page a nunjucks filter can be used that we've created to take the raw date from the payload and format it in a user friendly way whilst keeping the date formatting consistent across the service.

This can be used by changing `{{ appealDeadline }}` to `{{ appealDeadline | date }}`.

You can see an example of where this has been used in another page at https://github.com/Planning-Inspectorate/appeal-planning-decision/blob/master/packages/forms-web-app/src/views/eligibility/decision-date-passed.njk

**Review by shilpabasu22 — APPROVED** [review]
LGTM

**Review by nssdesikan — APPROVED** [review]

**Pull request merged**